### PR TITLE
ADEN-5701 added wait method

### PR DESCRIPTION
**Pull request by bulaszenko**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/361c575335cad9631cf635e4796192e13cc403af/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java [hunk 1]
```diff
@@ -42,28 +48,23 @@ public void adsRecoveryPageFairOasis(Page page) {
...
 
   @Test(
-      dataProviderClass = AdsDataProvider.class,
-      dataProvider = "adsRecoveryPageFairOasis",
       groups = "AdsRecoveryNoAdblockPageFairOasis"
   )
-  public void adsRecoveryNoAdblockPageFairOasis(Page page) {
-    String url = urlBuilder.getUrlForPage(page);
+  public void adsRecoveryNoAdblockPageFairOasis() {
+    System.out.println(RECOVERY_ARTICLE);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS106)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/e3e5a7bfe3ae3e1c13ee358dc1339ad50169106a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/c74733d297233152b86b37ad7057cc12bda668db/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/c4bd0d7de9a748c4e9483f40b663ef173069e12b/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java [hunk 2]
```diff
@@ -19,15 +17,23 @@
 
   private static Dimension DESKTOP_SIZE = new Dimension(1920, 768);
 
+  private static final String WIKIA = "arecovery";
+  private static final String WIKIA_ARTICLE = "SyntheticTests/Static_image";
+  private static final String INSTART_LOGIC_SWITCHED_OFF = "InstantGlobals.wgAdDriverInstartLogicRecoveryCountries=[ZZ]";
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java]: If you want to be consistent in your naming: rename `INSTART_LOGIC_SWITCHED_OFF` to `INSTANT_GLOBAL_INSTART_LOGIC_SWITCHED_OFF` or resign from `INSTANT_GLOBAL_*` prefix in other statics, please :)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java [hunk 3]
```diff
@@ -19,15 +17,23 @@
...
+  private static final String INSTANT_GLOBAL_PREMIUM_AD_LAYOUT_SWITCHED_OFF = "InstantGlobals.wgAdDriverPremiumAdLayoutCountries=[ZZ]";
+  private static final String INSTANT_GLOBAL_PAGE_FAIR_SWITCHED_ON = "InstantGlobals.wgAdDriverPageFairRecoveryCountries=[XX]";
+
+  private static final String RECOVERY_ARTICLE = WIKIA_ARTICLE + "?" + INSTANT_GLOBAL_PREMIUM_AD_LAYOUT_SWITCHED_OFF +
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java]: Consider building this in a new method (which you re-use in the test methods) and using `format()` in order to make it more readable.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/d6f72b480849fd9a0af1160cad428380e1837e19/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java [hunk 4]
```diff
@@ -42,30 +43,28 @@ public void adsRecoveryPageFairOasis(Page page) {
...
+  public void adsRecoveryLoggedInPageFairOasis() {
+    String url = urlBuilder.getUrlForPath(WIKIA, articlePageFairRecoveryUrl());
     AdsRecoveryObject adsRecoveryObject = new AdsRecoveryObject(driver, url, DESKTOP_SIZE);
-    adsRecoveryObject.refreshPageAddingCacheBuster();
 
     adsRecoveryObject.verifyNumberOfPageFairRecoveredSlots(0);
   }
-}
 
+  private String articlePageFairRecoveryUrl() {
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsPageFairRecoveryOasis.java]: Try to use [com.wikia.webdriver.common.core.url.UrlBuilder#appendQueryStringToURL](https://github.com/Wikia/selenium-tests/blob/346ed11190a0fbbd808406eb8dad6b433f9d7b7b/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L139) except this

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/3d6edad56d0ab2f6b6cb7304a6066ddd99dbf21d/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/5df24f5729c3ce2e4f1ca3526211d981f0f974bd/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsRecoveryObject.java#L45](https://github.com/Wikia/selenium-tests/blob/e48720fe650db68955f0eb0a24479a61eac7c2a7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java#L45): Define a constant instead of duplicating this literal "background" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**